internal/config/validate.go

```diff
@@ -48,6 +47,33 @@ type validateVariable struct {
 	Description string    `hcl:"description,optional"`
 }
 
+type ValidationResult struct {
+	Error   error
+	Warning string
+}
+
+func (v ValidationResult) String() string {
+	if v.Error != nil {
+		return v.Error.Error()
+	}
+
+	return "warning: " + v.Warning
+}
+
+type ValidationResults []ValidationResult
+
+func (v ValidationResults) Error() string {
+	var values []string
+
+	for _, res := range v {
+		values = append(values, res.String())
+	}
+
+	return fmt.Sprintf("%d validation errors: %s", len(v), strings.Join(values, ", "))
+}
+
+const AppeningHappening = "More than one app stanza within a waypoint.hcl file is deprecated, and will be removed in 0.10.\nPlease see https://discuss.hashicorp.com/t/deprecating-projects-or-how-i-learned-to-love-apps/40888 for more information."
```

Comment:
should we make this more obnoxious? like...
```suggestion
const AppeningHappening = "[DEPRECATION] More than one app stanza within a waypoint.hcl file is deprecated, and will be removed in 0.10.\nPlease see https://discuss.hashicorp.com/t/deprecating-projects-or-how-i-learned-to-love-apps/40888 for more information."
```